src/snapred/backend/data/GroceryService.py

```diff
@@ -343,6 +352,14 @@ def fetchGroceryList(self, groceryList: List[GroceryListItem]) -> List[Workspace
                     item.instrumentPropertySource = "InstrumentDonor"
                     item.instrumentSource = prev
                 res = self.fetchGroupingDefinition(item)
+            # for output (i.e. special-order) workspaces
+            elif item.workspaceType == "diffcal_output":
+                res = {"result": True, "workspace": self._createDiffcalOutputWorkspaceName(item.runNumber)}
+            elif item.workspaceType == "diffcal_table":
+                res = {"result": True, "workspace": self._createDiffcalTableWorkspaceName(item.runNumber)}
+            elif item.workspaceType == "diffcal_mask":
+                res = {"result": True, "workspace": self._createDiffcalMaskWorkspaceName(item.runNumber)}
+
```

Comment:
I would like to see this elif chain refactored to a dict of callables at some point but wont burden this pr with it.

Reply:
Ironically, I had that same thought, but was trying to be consistent with the existing implementation.  :(

Reply:
I ended up changing these to "match / case" statements.